cranelift/peepmatic/examples/preopt.peepmatic

```diff
@@ -0,0 +1,193 @@
+;; Apply basic simplifications.
+;;
+;; This folds constants with arithmetic to form `_imm` instructions, and other
+;; minor simplifications.
+;;
+;; Doesn't apply some simplifications if the native word width (in bytes) is
+;; smaller than the controlling type's width of the instruction. This would
+;; result in an illegal instruction that would likely be expanded back into an
+;; instruction on smaller types with the same initial opcode, creating
+;; unnecessary churn.
+
+;; Binary instructions whose second argument is constant.
+(=> (when (iadd $x $C)
+      (fits-in-native-word $C))
+    (iadd_imm $C $x))
```

Comment:
Why are the constants here at the beginning, instead of at the end like in the real clif ir?

Reply:
In clif, some immediates are at the beginning (e.g. condition codes) and some are at the end (e.g. constant values). For simplicity, I implemented peepmatic such that all immediates come first, followed by all operands. There is no inherent reason it needs to be this way.

Reply:
This order is confusing for `isub_imm`.